l1-contracts/contracts/bridge/AssetRouterBase.sol

```diff
@@ -0,0 +1,268 @@
+    function _requestToBridge(
+        // solhint-disable-next-line no-unused-vars
+        uint256 _chainId,
+        address _prevMsgSender,
+        bytes32 _assetId,
+        bytes memory _bridgeMintCalldata,
+        bytes32 _txDataHash,
+        bool _deposit
+    ) internal view virtual returns (L2TransactionRequestTwoBridgesInner memory request) {
+        bytes memory l2TxCalldata = abi.encodeCall(
+            IAssetRouterBase.finalizeTransfer,
+            (_chainId, _assetId, _bridgeMintCalldata)
+        );
+
+        request = L2TransactionRequestTwoBridgesInner({
+            magicValue: TWO_BRIDGES_MAGIC_VALUE,
+            l2Contract: L2_ASSET_ROUTER_ADDR,
+            l2Calldata: l2TxCalldata,
+            factoryDeps: new bytes[](0),
+            txDataHash: _txDataHash
+        });
+    }
+
+    /// @notice Forwards the burn request for specific asset to respective asset handler.
+    /// @param _chainId The chain ID of the ZK chain to which to deposit.
+    /// @param _value The L2 `msg.value` from the L1 -> L2 deposit transaction.
+    /// @param _assetId The deposited asset ID.
+    /// @param _prevMsgSender The `msg.sender` address from the external call that initiated current one.
+    /// @param _transferData The encoded data, which is used by the asset handler to determine L2 recipient and amount. Might include extra information.
+    /// @return bridgeMintCalldata The calldata used by remote asset handler to mint tokens for recipient.
+    function _burn(
+        uint256 _chainId,
+        uint256 _value,
+        bytes32 _assetId,
+        address _prevMsgSender,
+        bytes memory _transferData
+    ) internal returns (bytes memory bridgeMintCalldata) {
+        address l1AssetHandler = assetHandlerAddress[_assetId];
+        bridgeMintCalldata = IAssetHandler(l1AssetHandler).bridgeBurn{value: msg.value}({
+            _chainId: _chainId,
+            _mintValue: _value,
+            _assetId: _assetId,
+            _prevMsgSender: _prevMsgSender,
+            _data: _transferData
+        });
+    }
+
+    /// @notice Transfers allowance to Native Token Vault, if the asset is registered with it. Does nothing for ETH or non-registered tokens.
+    /// @dev assetId is not the padded address, but the correct encoded ID (NTV stores respective format for IDs).
+    /// @param _assetId The encoding of asset ID.
+    /// @param _amount The asset amount to be transferred to native token vault.
+    /// @param _prevMsgSender The `msg.sender` address from the external call that initiated current one.
+    function _transferAllowanceToNTV(bytes32 _assetId, uint256 _amount, address _prevMsgSender) internal {
+        address tokenAddress = nativeTokenVault.tokenAddress(_assetId);
+        if (tokenAddress == address(0) || tokenAddress == ETH_TOKEN_ADDRESS) {
+            // ToDo: Is it ok to keep the second check in base contract?
```

Comment:
Here we'll probably rename ETH_TOKEN_ADDRESS to BASE_TOKEN_ADDRESS and pass it in constructor as you suggested

Reply:
done